src/modules/deal/dto/deals-request.dto.ts

```diff
@@ -0,0 +1,73 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsDateString, IsNotEmpty, IsNumber, IsOptional, IsString, MaxLength } from 'class-validator';
+
+export class CreateDealDto {
+  @ApiProperty({
+    example: '0020900035', // Comment on PR if we need a description when it's self evident
+  })
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(10)
+  dealIdentifier: string;
+
+  @ApiProperty({
+    description: 'Deal portfolio is E1',
+    example: 'E1',
+  })
+  @IsString()
+  @IsOptional()
+  @MaxLength(2)
+  portfolioIdentifier = 'E1';
+
+  @ApiProperty({
+    description: 'Deal currency code is USD, CAD',
+    example: 'GBP',
+  })
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(3)
+  currency: string;
+
+  @ApiProperty()
+  @ApiProperty({
+    example: 2000000000.9989765,
+  })
+  @IsNumber()
+  @IsNotEmpty()
+  dealValue: number;
+
+  @ApiProperty({
+    description:
+      'Deal effective date is not in workflow. Currently operations assume a date earlier than the Facility effective date. Use the earliest Effective Date on facilities in Workflow under this Deal, i.e. Guarantee Commencement Date.',
+    example: '2017-06-01',
+  })
+  @IsDateString()
+  @IsNotEmpty()
```

Comment:
Do we need ISO Date validation?

Reply:
IsDateString is alias for IsISO8601 validator. In MDM-API we used IsDateString